prov/rxm/src/rxm.h

```diff
@@ -956,6 +957,29 @@ static inline int rxm_cq_write_recv_comp(struct rxm_rx_buf *rx_buf,
 				    rx_buf->pkt.hdr.tag);
 }
 
+static inline enum fi_hmem_iface rxm_iov_to_hmem_iface(const struct iovec *iov,
+						       size_t count)
+{
+	if (!count || !iov)
+		return FI_HMEM_SYSTEM;
+	return ofi_get_hmem_iface(iov[0].iov_base);
```

Comment:
This is a huge overhead to add to every send call.

Reply:
Yep. But to support FI_HMEM without FI_MR_HMEM, seems like we need to do HMEM iface discovery.

Reply:
This is only supporting this option when rxm is being used.  The check is more generic than that provider.  It more directly applies to verbs than rxm.  But it also applies to rxd and efa and probably out of tree providers.  We're heading down the path of adding checks and branches to every possible data transfer path in every provider in order to handle hmem.  There has to be a better way to do this, especially since we expanded the API to allow the user to indicate if hmem is being used or not.

ofiwg discussions talked about using a hook provider to support applications that want hmem support without api changes.

Reply:
I'll have some conversations internally about a hooking provider and get back to you.

Reply:
Talked it over interally and a hooking provider should work. I will remove the iface discovery commit from this PR. In addition, I am going to create a new hooking provider issue to work through the design of it since there are some components of it which are unclear.